Makefile

```diff
@@ -43,31 +43,48 @@ build:
 PLATFORMS := darwin linux windows
 
 GO_LDFLAGS_WIN = -ldflags "-X github.com/mendersoftware/mender-artifact/cli.Version=$(VERSION) -linkmode=internal -s -w -extldflags '-static' -extld=x86_64-w64-mingw32-gcc"
+GOARCH = "amd64"
 
-build-natives:
-	@arch="amd64";
-	@echo "building mac";
-	@env GOOS=darwin GOARCH=$$arch CGO_ENABLED=0 \
-		$(GO) build -a $(GO_LDFLAGS) $(BUILDV) $(BUILDTAGS) -o $(PKGNAME)-darwin ;
+build-natives-linux:
 	@echo "building linux";
-	@env GOOS=linux GOARCH=$$arch \
+	@env GOOS=linux GOARCH=$(GOARCH) \
 		$(GO) build -a $(GO_LDFLAGS) $(BUILDV) $(BUILDTAGS) -o $(PKGNAME)-linux ;
+
+build-natives-windows:
 	@echo "building windows";
-	@env GOOS=windows GOARCH=$$arch CC=x86_64-w64-mingw32-gcc CXX=x86_64-w64-mingw32-g++ \
+	@env GOOS=windows GOARCH=$(GOARCH) CC=x86_64-w64-mingw32-gcc CXX=x86_64-w64-mingw32-g++ \
 		$(GO) build $(GO_LDFLAGS_WIN) $(BUILDV) -tags $(TAGS) nolzma -o $(PKGNAME)-windows.exe ;
 
+build-natives-darwin:
+	@echo "building mac";
+	@env GOOS=darwin GOARCH=$(GOARCH) CGO_ENABLED=0 \
+		$(GO) build -a $(GO_LDFLAGS) $(BUILDV) $(BUILDTAGS) -o $(PKGNAME)-darwin ;
+
 build-contained:
 	rm -f mender-artifact && \
 	image_id=$$(docker build -f Dockerfile -q .) && \
 	docker run --rm --entrypoint "/bin/sh" -v $(shell pwd):/binary $$image_id -c "cp /go/bin/mender-artifact /binary" && \
 	docker image rm $$image_id
 
-build-natives-contained:
-	rm -f mender-artifact-darwin mender-artifact-linux mender-artifact-windows.exe && \
-	image_id=$$(docker build -f Dockerfile.binaries -q .) && \
+build-natives-contained-linux:
+	rm -f mender-artifact-linux && \
+	image_id=$$(docker build -f Dockerfile.binaries -q --build-arg PLATFORM=linux . ) && \
 	docker run --rm --entrypoint "/bin/sh" -v $(shell pwd):/binary $$image_id -c "cp /go/bin/mender-artifact* /binary" && \
 	docker image rm $$image_id
 
+build-natives-contained-darwin:
+	rm -f mender-artifact-darwin && \
+	image_id=$$(docker build -f Dockerfile.binaries -q --build-arg PLATFORM=darwin . ) && \
+	docker run --rm --entrypoint "/bin/sh" -v $(shell pwd):/binary $$image_id -c "cp /go/bin/mender-artifact* /binary" && \
+	docker image rm $$image_id
+
+build-natives-contained-windows:
+	rm -f mender-artifact-windows.exe && \
+	image_id=$$(docker build -f Dockerfile.binaries -q --build-arg PLATFORM=windows . ) && \
+	docker run --rm --entrypoint "/bin/sh" -v $(shell pwd):/binary $$image_id -c "cp /go/bin/mender-artifact* /binary" && \
+	docker image rm $$image_id
+
```

Comment:
Can we revert all this? I don't really understand the purpose (we already have a mechanism to cross-compile, right?). I suspect that changing this will trigger problems in mender-qa and/or meta-mender.

Reply:
by "all this" I mean changes in files `Makefile`, `Dockerfile.binaries` and `gitlab-ci.yml` (the parts related to the new make targets).

Reply:
I did not understand why all binaries were built in one stage - a rule of thumb is to split operations in CI and don't group them unnecessarily. From my perspective, it's better to see which platform builds correctly and which is not right away.

Reply:
As discussed, I'll create a separate PR with mentioned CI refactor.

Reply:
> a rule of thumb is to split operations in CI and don't group them unnecessarily. From my perspective, it's better to see which platform builds correctly and which is not right away.

The build is one step in CI and should work at all times for all platforms, if any platform build fails you will still see this in the job + you would see if a platform fails in the following tests.
By building all platforms in one step you can use the cross compilation benefit that go as a language brings and reuse the build environment.